Libplanet/Blockchain/BlockChain.cs

```diff
@@ -511,29 +511,24 @@ public Transaction<T> GetTransaction(TxId txId)
             Append(block, currentTime, evaluateActions: true, renderActions: true);
 
         /// <summary>
-        /// Adds <paramref name="transactions"/> to the pending list so that
-        /// a next <see cref="Block{T}"/> to be mined contains these
-        /// <paramref name="transactions"/>.
+        /// Adds <paramref name="transaction"/> to the pending list so that
+        /// a next <see cref="Block{T}"/> to be mined contains given
+        /// <paramref name="transaction"/>.
         /// </summary>
-        /// <param name="transactions"><see cref="Transaction{T}"/>s to add to the pending list.
+        /// <param name="transaction"><see cref="Transaction{T}"/> to add to the pending list.
         /// </param>
-        public void StageTransactions(IImmutableSet<Transaction<T>> transactions)
+        /// <exception cref="InvalidTxException">Thrown when the given
+        /// <paramref name="transaction"/> is invalid. </exception>
+        public void StageTransaction(Transaction<T> transaction)
         {
+            // FIXME it's global chain lock so using it in this method can cause degrading
+            // parallelism of `BlockChain<T>`. we should re-organize locks in `BlockChain<T>`
             _rwlock.EnterWriteLock();
 
             try
             {
-                // FIXME: We need to create a separate pool to handle transactions.
-                transactions = transactions
-                    .Where(tx => !_transactions.ContainsKey(tx.Id))
-                    .ToImmutableHashSet();
-
-                foreach (Transaction<T> tx in transactions)
-                {
-                    _transactions[tx.Id] = tx;
-                }
-
-                Store.StageTransactionIds(transactions.Select(tx => tx.Id).ToImmutableHashSet());
+                _transactions[transaction.Id] = transaction;
+                Store.StageTransactionIds(new[] { transaction.Id }.ToImmutableHashSet());
```

Comment:
Using [`ImmutableHashSet.Create<T>(T)` method][1] might be slightly more efficient for a single element.

```suggestion
                Store.StageTransactionIds(ImmutableHashSet.Create<TxId>(transaction.Id));
```

[1]: https://docs.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutablehashset.create?view=netcore-3.1&viewFallbackFrom=netstandard-2.1#System_Collections_Immutable_ImmutableHashSet_Create__1___0_